Next version

Keats opened the pull request.

DerSaidin committed:
Add slugify.paths_keep_dates option. Update docs.

Manual testing:

Existing test_site/config.toml

http://127.0.0.1:1111/posts/a-post-with-dates

====

--- a/test_site/config.toml
+++ b/test_site/config.toml
@@ -18,6 +18,7 @@ extra_syntaxes_and_themes = ["syntaxes", "highlight_themes"]

 [slugify]
 paths = "on"
+paths_keep_dates = true
 taxonomies = "on"
 anchors = "on"

http://127.0.0.1:1111/posts/2016-10-08-a-post-with-dates

====

--- a/test_site/config.toml
+++ b/test_site/config.toml
@@ -17,7 +17,8 @@ highlight_theme = "custom_gruvbox"
 extra_syntaxes_and_themes = ["syntaxes", "highlight_themes"]

 [slugify]
-paths = "on"
+paths = "off"
+paths_keep_dates = true
 taxonomies = "on"
 anchors = "on"

http://127.0.0.1:1111/posts/2016-10-08_a-post-with-dates

====

--- a/test_site/config.toml
+++ b/test_site/config.toml
@@ -17,7 +17,8 @@ highlight_theme = "custom_gruvbox"
 extra_syntaxes_and_themes = ["syntaxes", "highlight_themes"]

 [slugify]
-paths = "on"
+paths = "safe"
+paths_keep_dates = true
 taxonomies = "on"
 anchors = "on"

http://127.0.0.1:1111/posts/2016-10-08_a-post-with-dates

Keats committed:
Next version

Newbytee committed:
Metainfo: Add 0.16.1

JoelMon committed:
Imporved is_colocated_asset_links (#1972)

This PR includes the following:

- Added a check for `../` which would check to see if a file is in a
  _sub_ directory
- Wrote test for `is_colocated_asset_links`

This PR is similare to PR #1969

tranzystorekk committed:
Add completion subcommand (#1968)

* Add completion subcommand

* Remove old completion scripts

* Move completion logic

scouten committed:
Get term fn (#1964)

* Add new `get_taxonomy_term` global function

* Update per @Keats feedback to make `page_count` available all the time.

Keats committed:
Simplify taxonomy term struct

bemyak committed:
Add sorting by slug (#1926)

ZettaScript committed:
Tera date-locale feature (#1978)

Yaroslav-95 committed:
Add flag to override prompt for existing dir (#1986)

* Add flag to override prompt for existing dir

Added --force flag to build subcommand.

* Remove unnecessary 'force' check

* Remove prompt to overwrite directory on build cmd

Keats committed:
Update deps

Keats committed:
Handle colocated links to assets

Solves #1993

Hoverbear committed:
Use BufWriter in imageproc (#1996)

According to
https://docs.rs/image/0.24.3/image/enum.DynamicImage.html#method.write_to,
it assumes a buffered writer, so we should provide one.

Signed-off-by: Ana Hobden <operator@hoverbear.org>

Commit:
add get_hash function (#1962)



Co-authored-by: Newbyte <newbie13xd@gmail.com>
Co-authored-by: n-tux-tmp <n-tux-tmp@example.com>
Co-authored-by: n-tux-tmp <n-tux-tmp@noreply.example.org>

Commit:
Update changelog

JeremyBarbosa committed:
Minify embedded css and js with additional lib flags (#1922)

Commit:
Fix test

iEverX committed:
Fix current_url in taxonomy term (#2000)

* Fix current_url in taxonomy term

* add tests

ffrizzo committed:
Add new flag 'port_append' to give the ability to remove port from base url (#2003)

Keats committed:
Fix no-port-append default

sgued committed:
Fix clippy::derive_partial_eq_without_eq

sgued committed:
Site: fix clippy warnings

sgued committed:
templates: Fix clippy warning

sgued committed:
content: Fix clippy warnings

sgued committed:
imageproc: Fix clippy warning

sgued committed:
markdown: Fix clippy warnings

sgued committed:
Fix clippy::manual_retain

sgued committed:
Add clippy to CI

mscherer committed:
Do not overfilter HTML when removing footnotes (#2008)

If the summary contain HTML using <sup> like "the 1<sup>st</<sup>", it get
filtered as the regexp that remove footnotes is not precise
enough to remove just the footnotes.

sgued committed:
Search json index (#1998)

* search: Add support for a JSON index

* docs: Document JSON index for search

* docs: Use lazy-loaded JSON index

* Add elasticlunr prefix to search engine format configuration

This will be useful if support for more search libraries are added in the future

mscherer committed:
Filter more than 1 footnote in summary (#2017)

If the summary contains more than 1 footnote, only the 1st is
removed, and so dangling links exists for all the others.

sgued committed:
Cargo fmt (#2009)

* Run cargo fmt

* Add cargo fmt --check to pipeline

sgued committed:
Fix needless_borrow warnings

flxzt committed:
add preserve_dotfiles_in_output configuration option (#1985)

magikstm committed:
Small update to the docs for get_url (#2033)

Small update to the doc linked to the discussion on #2026.

magikstm committed:
Add details in the doc on CommonMark extension (#2034)

* Add details in the doc on CommonMark extension

* Update overview.md

davidmreed committed:
Expose `markdown` config property to templates (#2052)

* Expose the markdown config property to templates

* Document additional strategies for linking code-highlight stylesheets

Keats committed:
Update some deps

Keats committed:
Replace libsass by grass

Closes #1535
Closes #2022

Keats committed:
Fmt + change min version

Keats committed:
Clippy + 1.63

chris-morgan commented on CHANGELOG.md:
```diff
@@ -1,5 +1,26 @@
+- Add new flag 'port_append' to give the ability to remove port from base url
+- `config.markdown` is now available in templates
+- Add `preserve_dotfiles_in_output` option in the config
+- Replace libsass by a Rust implementation: [grass](https://github.com/connorskees/grass)
```
This is a breaking change:

> grass is not a drop-in replacement for libsass and does not intend to be. If you are upgrading to grass from libsass, you may have to make modifications to your stylesheets, though these changes should not differ from those you would have to make if upgrading to dart-sass.

Keats replied:
I'll move it, i forgot some stuff like division changed.

Keats committed:
libsass -> grass is a breaking change

Keats committed:
Add upgrade guide for Sass

Keats committed:
Update changelog

Keats committed:
Update base64

vcrn committed:
Merge default language data with section language data (#2039)

* Merge language data in config.toml

Keats committed:
Use macro for LanguageOptions::merge

Keats committed:
Update changelog

Keats committed:
Remove code blocks from search index

Kinda closes #1176

orlp committed:
Reduced cachebust fingerprint to be more reasonable (#2074)

* Reduced cachebust fingerprint to be more reasonable.

Keats committed:
fmt + changelog + updates

Keats committed:
Add plain text syntax alias to text

Closes #1633

Keats committed:
Use reqwest rustls-tls

Easier cross-compiling
Closes #1569

Keats committed:
Remove built-in shortcodes

Bazooka way of fixing #1832

Keats committed:
Revert "Use reqwest rustls-tls"

This reverts commit 17c8aa77a5eb198853bcd63a9384c4a8ceba053c.

Keats committed:
Handle pages named index.md in sections

Closes #2082

Keats committed:
Filter out temp files or with no extensions from colocated assets

Keats committed:
Errors on empty taxonomies term

Closes #2085

orlp committed:
Removed faulty Windows-specific path check/conversion. Fixes #2076. (#2088)

Keats committed:
Procimage refactor (#2086)

* Refactor image proc

Closes #2066

* Add colocated_path to shortcodes

Closes #1793

Keats committed:
clippy

Keats committed:
Bump Rust CI version

Keats committed:
Update toml lib

Closes #1313

Keats committed:
Try to disable uninlined format lint

sethm committed:
Page and config authors (#2024) (#2092)

The W3C feed validator fails to validate RSS 2.0 and Atom 1.0 feed
elements that do not contain a valid author. This change adds an
`authors: Vec<String>` to pages, as well as an `author: Option<String>`
to Config that will act as a default to use in RSS and Atom templates if
no page-level authors are specified.

kartva committed:
feat: External URL redirect (#2080)

Keats committed:
Update changelog & mockito

Keats committed:
Downgrade mockito to not raise MSRV

Keats committed:
Add test for #1566

Closes #1566

Keats committed:
cargo fmt

Keats committed:
Remove println

pqcfox committed:
Allow external links to start with "www.*" (#2100)

* Make www.* URLs considered valid external links

* Tweak description of is_external_link

Keats committed:
Update changelog & test-case

Keats committed:
get ready for release
